sql/core/src/test/resources/sql-tests/inputs/group-by.sql

```diff
@@ -179,3 +179,8 @@ SELECT count(*) FROM test_agg WHERE k = 1 or k = 2 or count(*) + 1L > 1L or max(
 
 -- Aggregate with multiple distinct decimal columns
 SELECT AVG(DISTINCT decimal_col), SUM(DISTINCT decimal_col) FROM VALUES (CAST(1 AS DECIMAL(9, 0))) t(decimal_col);
+
+-- SPARK-34581: Don't optimize out grouping expressions from aggregate expressions without aggregate function
+SELECT not(a IS NULL), count(*) AS c
+FROM testData
+GROUP BY a IS NULL
```

Comment:
can we also test `select a + b + rand() .... group by a + b`？

Reply:
Added in https://github.com/apache/spark/pull/31913/commits/cff9b9ab289412529728749001ded9550f14e66e.